src/map/status.cpp

```diff
@@ -4130,8 +4174,12 @@ int status_calc_pc_sub(struct map_session_data* sd, enum e_status_calc_opt opt)
 	else if(pc_isridingdragon(sd))
 		base_status->aspd_rate += 250-50*pc_checkskill(sd,RK_DRAGONTRAINING);
 #else // Needs more info
-	if((skill = pc_checkskill(sd,SG_DEVIL)) > 0 && pc_is_maxjoblv(sd))
-		base_status->aspd_rate += 30*skill;
+	if ((skill = pc_checkskill(sd,SG_DEVIL)) > 0 && ((sd->class_&MAPID_THIRDMASK) == MAPID_STAR_EMPEROR || sd->status.job_level >= 50)) {
+		if ((skill = pc_checkskill(sd,SJ_PURIFY)) > 0)
+			base_status->aspd_rate += 30*skill;
+		else
+			base_status->aspd_rate += 15*skill;
+	}
```

Comment:
Variable of the skill value will be overwrite.
When SG_DEVIL = lv10 , SJ_PURIFY = lv1
So   += 30*skill     will be     30*1

SJ_DEVIL ASPD in Kro and TWro
only + 2~11%

And I search about SJ_PURIFY
Seems only reduced the visual punishment.
Didn't reduced the aspd bonus.

source-1: https://rd.fharr.com/skill-441.html
source-2: TWro skilldescription.lub
source-3: https://rd.fharr.com/skill-2589.html